File: taskflow_app/backend/app/models/task.py
# app/models/task.py
# Pydantic models for task request validation and API responses.

from pydantic import BaseModel
from datetime import datetime
from typing import Literal


class TaskCreate(BaseModel):
    title: str
    description: str | None = None
    priority: Literal["low", "medium", "high"] = "medium"
    deadline: datetime | None = None


class TaskUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    completed: bool | None = None
    priority: Literal["low", "medium", "high"] | None = None
    deadline: datetime | None = None


class TaskResponse(BaseModel):
    id: str
    owner_uid: str
    title: str
    description: str | None = None
    completed: bool
    priority: str
    deadline: str | None = None
    created_at: str

File: frontend/lib/features/tasks/domain/entities/task.dart
import 'package:equatable/equatable.dart';
import 'task_priority.dart';
import 'task_status.dart';

class Task extends Equatable {
  final String id;
  final String title;
  final String description;
  final TaskStatus status;
  final TaskPriority priority;
  final DateTime deadline;
  final DateTime createdAt;

  const Task({
    required this.id,
    required this.title,
    this.description = '',
    this.status = TaskStatus.pending,
    this.priority = TaskPriority.medium,
    required this.deadline,
    required this.createdAt,
  });

  @override
  List<Object?> get props => [id, title, description, status, priority, deadline, createdAt];
}
